components/brave_wallet/browser/internal/orchard_storage/orchard_storage.cc

```diff
@@ -1058,11 +1093,37 @@ ZCashOrchardStorage::AddCheckpoint(const mojom::AccountIdPtr& account_id,
                                   database_.GetErrorMessage()});
   }
 
-  return base::ok(true);
+  return base::ok(Result::kSuccess);
+}
+
+base::expected<OrchardStorage::Result, OrchardStorage::Error>
+OrchardStorage::UpdateCheckpoint(const mojom::AccountIdPtr& account_id,
+                                 uint32_t checkpoint_id,
+                                 const OrchardCheckpoint& checkpoint) {
+  base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
+                                                base::BlockingType::WILL_BLOCK);
+  auto get_checkpoint_result = GetCheckpoint(account_id, checkpoint_id);
+  RETURN_IF_ERROR(get_checkpoint_result);
+  if (!get_checkpoint_result.value()) {
+    return base::ok(Result::kFailure);
```

Comment:
This doesn't seem quite right to me, and I'm glad we have changed to an enum, as it has made the whole thing more apparent. We have this return type:

```cxx
base::expected<OrchardStorage::Result, OrchardStorage::Error>
```

So for the success path we are supposed to have `Result`, and for the failure path we have `Error`, however also have `Result::kFailure`, which seems to be misplaced here, as this arguably should be part of the `Error` path. Would you agree? If that's not the case, then maybe `kFailure` should have some other name.

Reply:
>which seems to be misplaced here, as this arguably should be part of the Error path
I think Error means some internal storage error while result could have no effect due wrong inputs for ex.
I'll rename it to kNone